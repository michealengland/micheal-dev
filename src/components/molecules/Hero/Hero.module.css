.hero {
  @apply relative flex flex-col items-center justify-center font-sans bg-primary-purple1 text-nuetral-white text-center h-auto max-h-full max-w-full z-1 p-4 mb-4;

  background-position: center;
  background-size: cover;
  clip-path: polygon(0 0, 100% 0, 100% 88%, 50% 100%, 0 88%, 0% 25%);
  min-height: 60vh;

  @media screen and (min-width: 48em) {
    @apply py-4 px-6;
  }

  &::before {
    @apply absolute z-behind inset-0 bg-primary-purple1 bg-opacity-80;

    content: '';
  }

  & .title {
    @apply text-5xl md:text-6xl lg:text-7xl font-semibold;
  }

  & .content {
    @apply text-lg lg:text-xl font-semibold mt-6;
  }

  & .inner-wrap {
    @apply relative bg-primary-purple2 z-1 my-0 mx-auto text-left w-full py-12 px-5 max-w-2xl;

    @screen lg {
      @apply p-8;
    }
  }
}
