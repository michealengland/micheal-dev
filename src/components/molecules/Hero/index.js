import PropTypes from 'prop-types'
import React from 'react'
import styles from './Hero.module.css'
import Avatar from '@content/assets/mle-12-05-21-512x512.jpeg'

export default function Hero({title, content, img}) {
  const backgroundImage = img && {backgroundImage: `url(${img})`}

  return (
    <section className={styles.hero} style={backgroundImage}>
      <div>
        <img
          alt="Micheal England"
          className={styles.avatar}
          height={512}
          src={Avatar}
          width={512}
        />
        {title && <h1 className={styles.title}>{ title }</h1>}
        {content && <p className={styles.content}>{ content }</p>}
      </div>
    </section>
  )
}

Hero.propTypes = {
  content: PropTypes.string,
  img: PropTypes.string,
  title: PropTypes.string,
}
