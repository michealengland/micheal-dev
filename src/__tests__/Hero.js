import React from 'react'
import renderer from 'react-test-renderer'
import Hero from '../components/molecules/Hero'
describe('Hero', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Hero
          content={'Hero subtitle.'}
          img={'./images/wcus2018-hall-track-1.jpg'}
          title={'Hero title.'}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
