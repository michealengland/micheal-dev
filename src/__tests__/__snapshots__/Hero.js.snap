// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
<section
  className="hero"
  style={
    Object {
      "backgroundImage": "url(./images/wcus2018-hall-track-1.jpg)",
    }
  }
>
  <div>
    <h1
      className="title"
    >
      Hero title.
    </h1>
    <p
      className="content"
    >
      Hero subtitle.
    </p>
  </div>
</section>
`;
